api/src/main/java/ai/djl/ndarray/NDArray.java

```diff
@@ -581,14 +581,26 @@ default NDArray get(NDManager manager, long... indices) {
     NDArray gather(NDArray index, int axis);
 
     /**
-     * Returns a partial {@code NDArray} pointed by the indexed array according to linear indexing,
-     * and the of output is of the same shape as index.
+     * Returns a partial {@code NDArray} pointed by index according to linear indexing, and the of
+     * output is of the same shape as index.
      *
-     * @param index picks the elements of an NDArray to the same position as index
+     * @param index picks the elements of an NDArray and output to the same entry as in index
      * @return the partial {@code NDArray} of the same shape as index
      */
     NDArray take(NDArray index);
 
+    /**
+     * Set the entries of {@code NDArray} pointed by index according to linear indexing, to be the
+     * numbers in data, which is of the same shape as index.
+     *
+     * @param index select the entries of an {@code NDArray}
+     * @param data numbers to assign to the indexed entries
+     * @return The NDArray with updated values
+     */
+    default NDArray put(NDArray index, NDArray data) {
+        throw new UnsupportedOperationException("Not implemented yet.");
```

Comment:
We usually don't put the Unsupported default in NDArray because it makes it hard to keep track of for the "full" engines. Put it in the NDArrayAdapter and the full engines that are missing it (should be MX and TF)